default.nix

```diff
@@ -118,7 +118,9 @@ let
       # is not set, it defaults to `src`.
     , root ? src
     , packageLock ? null
-    , npmCommands ? [ "npm install --loglevel verbose" ]
+      # Propagate --nodedir argument into npm install, as it fixes issue with
+      # compiling with node-gyp package
+    , npmCommands ? [ "npm install --loglevel verbose --nodedir=${pkgs.nodejs}/include/node" ]
```

Comment:
Interesting. I tend to use https://github.com/jtojnar/nixfiles/blob/e2bfef939aada764f5e07e74452ad12a889ff92c/pkgs/activitywatch/default.nix#L333-L341 but if this is sufficient this would be much nicer.

The only issue I expect is that people might also need to add this flag when overriding `npmCommands`. Is there a way to set this using environment variable or `npm config`?

Reply:
As far as I know, it is impossible. The way `npm install` works is when there are additional flags, it propagates them into scripts that it runs. Maybe it is possible to set default flags to npm, but I have no clue how.
By the way, I found out about this flag [here](https://github.com/nodejs/node-gyp/blob/bc47cd60b986eaa55a23050d8f72d1cc117bdba0/lib/configure.js#L37) if you want to check and see how it works.